src/bptl/work_units/brp/tasks.py

```diff
@@ -37,3 +38,49 @@ def perform(self) -> dict:
         is_above_age = None if leeftijd is None else leeftijd >= age
 
         return {"isAboveAge": is_above_age}
+
+
+@register
+class DegreeOfKinship(WorkUnit):
+    """
+    Fetches BRP API and returns flag if certain two people have a specified degree of kinship
+
+    Required process variables:
+
+    * ``burgerservicenummer1``: BSN of the first person
+    * ``burgerservicenummer2``: BSN of the second person
+
+    The task sets the process variables:
+
+    * ``kinship``: integer, which represents the degree of kinship (blood relations). Values can be in range [1..4]
+    """
+
+    def perform(self) -> dict:
+        variables = self.task.get_variables()
+        bsn1 = variables["burgerservicenummer1"]
+        bsn2 = variables["burgerservicenummer2"]
+
+        if bsn1 == bsn2:
+            return {"kinship": None}
+
+        client = get_client_class()()
+        client.task = self.task
+
+        rel1 = Relations(bsn1)
+        rel2 = Relations(bsn2)
+
+        # 1. request 1-level relations
+        rel1.expand(client, 1)
+        rel2.expand(client, 1)
+
+        kinship = rel1.kinship(rel2)
+
+        if kinship:
+            return {"kinship": kinship}
+
+        # 2. Request 2-level relations
```

Comment:
If levels are tree-depths, then I would expect 4 levels (great great grand mother)? Tests seems to test for this so it probably works...